### PR TITLE
Added method for checking if a string matches an elements symbol or name

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
Adds a simple method to check if a string is the same as an elements name or its symbol.

### CODE REVIEW

**Comment by JensWehner** [comment]
can we change clang format to only format the code that was changed not the whole file? It is really hard to do code review.

**Comment by JoshuaSBrown** [comment]
I would not know how to do that. I can go ahead and format the hole repo. Then this should not be as difficult.

File: src/libtools/elements.cc [hunk 1]
```diff
@@ -148,7 +148,7 @@ bool Elements::isEleFull(std::string fullname) {
 }
 
 bool Elements::isEleShort(std::string shortname) {
-  if (!this - _filled_EleShort) {
+  if (!this->_filled_EleShort) {
```
Comment (JensWehner) [inline src/libtools/elements.cc]: I still do not get it, you should fill FillEleShort not long, this is confusing. I do not understand

Comment (JensWehner) [reply src/libtools/elements.cc]: I get the part with the function name, but the bools should not disagree

Comment (JoshuaSBrown) [reply src/libtools/elements.cc]: Oh, right this one is backwards.

File: src/libtools/elements.cc [hunk 2]
```diff
@@ -25,732 +26,767 @@ using namespace std;
...
-    }
-    try{
-        return _NucCrg.at(name);
-    }catch(const std::out_of_range& oor){
-        throw std::runtime_error("Nuclearcharge of element "+name+" not found.");
-    }
+bool Elements::isElement(std::string name) {
+  if (!this->_filled_EleShort) {
+    this->FillEleShort();
+    _filled_EleShort;
```
Comment (JensWehner) [inline src/libtools/elements.cc]: must be set to true

File: src/libtools/elements.cc [hunk 3]
```diff
@@ -25,732 +26,767 @@ using namespace std;
...
+  }
+  if (!this->_filled_EleFull) {
+    this->FillEleFull();
+    _filled_EleFull;
```
Comment (JensWehner) [inline src/libtools/elements.cc]: dito

**Comment by JoshuaSBrown** [comment]
@JensWehner Hey can you approve this?

File: src/libtools/elements.cc [hunk 4]
```diff
@@ -25,6 +26,19 @@ using namespace std;
 /*************************
  * Public Facing Methods *
  *************************/
+bool Elements::isElement(std::string name) {
+  if (!this->_filled_EleShort) {
+    this->FillEleShort();
+    _filled_EleShort = true;
+  }
+  if (!this->_filled_EleFull) {
```
Comment (JensWehner) [inline src/libtools/elements.cc]: call the other to methods below here instead of writing it again

**Comment by JoshuaSBrown** [comment]
@JensWehner I think I have now addressed your concerns.

**Pull request merged**